tests/integration/test_publish_on_linkedin.py

```diff
@@ -0,0 +1,254 @@
+# Standard Library
+from unittest import mock
+
+# Third Party Stuff
+import pytest
+import requests
+from django.conf import settings
+from rest_framework import status
+from tests import factories as f
+
+# nexus Stuff
+from nexus.base import exceptions as exc
+from nexus.social_media import services, tasks
+from nexus.social_media.models import Post
+
+pytestmark = pytest.mark.django_db
+
+
+@pytest.fixture
+def base_data(client):
+    post = f.create_post(image=None)
+    api_url_ugc = f"{settings.LINKEDIN_API_URL_BASE}ugcPosts"
+    linkedin = settings.LINKEDIN_AUTH
+    author = f"urn:li:organization:{linkedin['organization_id']}"
+    headers = {'X-Restli-Protocol-Version': '2.0.0',
+               'Content-Type': 'application/json',
+               'Authorization': f"Bearer {linkedin['access_token']}"}
+
+    post_data = {
+        "author": author,
+        "lifecycleState": "PUBLISHED",
+        "specificContent": {
+            "com.linkedin.ugc.ShareContent": {
+                "shareCommentary": {
+                    "text": post.text
+                },
+                "shareMediaCategory": "NONE"
+            },
+        },
+        "visibility": {
+            "com.linkedin.ugc.MemberNetworkVisibility": "PUBLIC"
+        },
+    }
+
+    return (post,
+            api_url_ugc,
+            linkedin,
+            author,
+            headers,
+            post_data,)
+
+
+@mock.patch('nexus.social_media.services.requests.post')
+def test_publish_on_linkedin_without_image(mock_post, base_data):
+    post, api_url_ugc, linkedin, author, headers, post_data = base_data
+
+    post_id = post.id
+
+    mock_post.return_value = requests.Response()
+
+    # check post only with text
+    response = services.publish_on_linkedin(post_id)
+    mock_post.assert_called_once_with(api_url_ugc, headers=headers,
```

Comment:
what is `ugc`?

Reply:
`ugc` actually refers to User Generated Content. To share on linkedIn, we send `POST` request to its `UGC` API.

Reply:
Please mention that in a comment.

Reply:
I have mentioned it in the `base_data` fixture